File: app/src/main/java/com/damhoe/scoresheetskat/app/app_settings/SettingsFragment.java
package com.damhoe.scoresheetskat.app.app_settings;

import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.damhoe.scoresheetskat.MainActivity;
import com.damhoe.scoresheetskat.R;
import com.damhoe.scoresheetskat.databinding.FragmentSettingsBinding;
import com.damhoe.scoresheetskat.shared_ui.utils.InsetsManager;

import java.util.Locale;

public class SettingsFragment extends Fragment {

    private SettingsViewModel viewModel;
    private FragmentSettingsBinding binding;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Resources res = getResources();
        Configuration config = res.getConfiguration();
        config.setLocale(new Locale("de"));
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater,
                R.layout.fragment_settings, container, false);

        // Inflate the preferences screen
        FragmentTransaction transaction = getParentFragmentManager().beginTransaction();
        transaction.replace(R.id.container, SettingsContentFragment.newInstance());
        transaction.commit();

        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        InsetsManager.applyStatusBarInsets(binding.appbarLayout);
        InsetsManager.applyNavigationBarInsets(binding.nestedScrollView);

        // Setup toolbar
        NavController navController = findNavController();
        AppBarConfiguration appBarConfiguration =
                ((MainActivity) requireActivity()).getAppBarConfiguration();
        NavigationUI.setupWithNavController(binding.toolbar, navController, appBarConfiguration);
    }

    private NavController findNavController() {
        return Navigation.findNavController(requireActivity(), R.id.nav_host_fragment);
    }
}

File: app/src/main/java/com/damhoe/skatscores/app/app_settings/SettingsFragment.kt
package com.damhoe.skatscores.app.app_settings

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController
import androidx.navigation.ui.NavigationUI.setupWithNavController
import androidx.preference.Preference
import androidx.preference.Preference.OnPreferenceClickListener
import com.damhoe.skatscores.R
import com.damhoe.skatscores.databinding.FragmentSettingsBinding
import com.damhoe.skatscores.shared_ui.utils.InsetsManager

class SettingsFragment : Fragment(), OnPreferenceClickListener {
    lateinit var binding: FragmentSettingsBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentSettingsBinding.inflate(inflater)

        // Inflate the preferences screen
        val transaction = parentFragmentManager.beginTransaction()
        transaction.replace(
            R.id.container,
            SettingsContentFragment.newInstance()
                .apply { setOnPreferenceClickListener(this@SettingsFragment) }
        )
        transaction.commit()

        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.run {
            InsetsManager.applyStatusBarInsets(appbarLayout)
            InsetsManager.applyNavigationBarInsets(nestedScrollView)

            // Setup toolbar
            val navController = findNavController()
            setupWithNavController(toolbar, navController)
        }
    }

    fun navigateToAboutFragment() {
        findNavController().navigate(SettingsFragmentDirections.actionAppSettingsToAbout())
    }

    private fun findNavController() =
        findNavController(requireActivity(), R.id.nav_host_fragment)

    override fun onPreferenceClick(preference: Preference): Boolean {
        if (preference.key == "about") {
            navigateToAboutFragment()
            return true
        }
        return false
    }
}